app/models/host/base.rb

```diff
@@ -469,6 +470,12 @@ def uniq_interfaces_identifiers
       success
     end
 
+    def build_managed_only
+      (!self.build? || self.managed?).tap do |valid|
```

Comment:
Why use `tap` and `unless valid` (line below) here?  This looks like it should be a simple conditional:

    if !managed? && build?
      errors.add(...)
    end